File: engine/b/src/main/java/net/crazyinvestor/engine_b/repository/TickerHistoryRepository.java
package net.crazyinvestor.engine_b.repository;

import net.crazyinvestor.engine_b.entity.Symbol;
import net.crazyinvestor.engine_b.entity.Ticker;
import net.crazyinvestor.engine_b.entity.TickerHistory;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TickerHistoryRepository extends JpaRepository<TickerHistory, Long> {
    Optional<TickerHistory> findByTicker(Ticker ticker);
}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/deprecated/repository/TickerHistoryRepository.kt
package net.crazyinvestor.ticker_producer_engine.deprecated.repository

import net.crazyinvestor.ticker_producer_engine.deprecated.entity.Ticker
import net.crazyinvestor.ticker_producer_engine.deprecated.entity.TickerHistory
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
class TickerHistoryRepository {
    fun findByTicker(ticker: Ticker): Optional<TickerHistory> {
        println("MOCK REPOSITORY4 $ticker")
        return Optional.of(
            TickerHistory(null, ticker, null,
            null,null,null,
            null,null,null,null)
        )
    }
    fun save(tickerHistory: TickerHistory): TickerHistory {
        println("save() MOCK REPOSITORY4 $tickerHistory")
        return tickerHistory
    }
}